internal/agent/install.go

```diff
@@ -113,6 +113,10 @@ func Install(dir ...string) error {
 		if exists {
 			tmpdir, err := os.MkdirTemp("", "kairos-install-")
 			if err == nil {
+				// Add cloud config header if necessary
+				if !strings.Contains(cloudConfig, "#cloud-config") {
+					cloudConfig = fmt.Sprintf("#cloud-config\n%s", cloudConfig)
```

Comment:
shouldn't we add the config in the config instead of here? otherwise the check for the config is not really that useful IMO

Reply:
This is the object that its marshalled into json and pumped via the event:

```
type CloudInit struct {
	Install       kairos.Install            `yaml:"install,omitempty" json:"install,omitempty"`
	Options       map[string]string         `yaml:"options,omitempty" json:"options,omitempty"`
	Cluster       clusterplugin.Cluster     `yaml:"cluster,omitempty" json:"cluster,omitempty"`
	Stages        map[string][]schema.Stage `yaml:"stages,omitempty" json:"stages,omitempty"`
	Stylus        Stylus                    `yaml:"stylus,omitempty" json:"stylus,omitempty"`
	StylusPackage *string                   `yaml:"stylusPackage,omitempty" json:"stylusPackage,omitempty"`
	Env           []string                  `yaml:"env,omitempty" json:"env,omitempty"`
}

```

So ti kinds of make sense that what we receive its not with the header and what we save as cloud config needs the header to be read back up.

I wonder if we could make the collector read dirs and Reader objects directly so we could apss that data directly and skip the header stuff for those files.

Reply:
Im wondering if we should just try to unmarshall the cloudConfig into an agentconfig.Config object directly and merge it on top of the base config which should be there already....

That would make this simpler and would make the event config end up on top of the others as expected....

Reply:
I don't remember how I used the mock provider back then but I assume I just copied that to [the directory where providers are expected](https://github.com/kairos-io/kairos-agent/blob/9f46af255f056c974d335dd305eac837fb012efb/internal/bus/bus.go#L35) and let kairos-agent run it.